examples/dtls-sock/dtls-client.c

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright (C) 2019 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     examples
+ * @{
+ *
+ * @file
+ * @brief       DTLS sock client example
+ *
+ * @author      Aiman Ismail <muhammadaimanbin.ismail@haw-hamburg.de>
+ */
+
+#include <stdio.h>
+
+#include "net/sock/udp.h"
+#include "net/sock/dtls.h"
+#include "net/ipv6/addr.h"
+#include "net/credman.h"
+
+#include "client_keys.h"
+
+#ifndef DTLS_DEFAULT_PORT
+#define DTLS_DEFAULT_PORT 20220 /* DTLS default port */
+#endif
+
+#define SOCK_DTLS_CLIENT_TAG (2)
+
+#ifdef DTLS_PSK
+static uint8_t psk_id_0[] = PSK_DEFAULT_IDENTITY;
+static uint8_t psk_key_0[] = PSK_DEFAULT_KEY;
+#endif /* DTLS_PSK */
+#ifdef DTLS_ECC
+static ecdsa_public_key_t other_pubkeys[] = {
+    { .x = other_pub_key_x, .y = other_pub_key_y },
+};
+#endif /* DTLS_ECC */
+
+static int client_send(char *addr_str, char *data, size_t datalen)
+{
+    uint8_t rcv[512];
+    ssize_t res;
+
+    sock_udp_t udp_sock;
+    sock_dtls_t dtls_sock;
+    sock_dtls_session_t session;
+    sock_udp_ep_t remote;
+    sock_udp_ep_t local = SOCK_IPV6_EP_ANY;
+    local.port = 12345;
+    remote.port = DTLS_DEFAULT_PORT;
+
+    /* get interface */
+    int iface = ipv6_addr_split_iface(addr_str);
+    if (iface >= 0) {
+        if (gnrc_netif_get_by_pid(iface) == NULL) {
+            puts("Invalid network interface");
+            return -1;
+        }
+        remote.netif = iface;
+    } else if (gnrc_netif_numof() == 1) {
+        /* assign the single interface found in gnrc_netif_numof() */
+        remote.netif = gnrc_netif_iter(NULL)->pid;
+    } else {
+        /* no interface is given, or given interface is invalid */
+        /* FIXME This probably is not valid with multiple interfaces */
+        remote.netif = SOCK_ADDR_ANY_NETIF;
+    }
+
+    if (!ipv6_addr_from_str((ipv6_addr_t *)remote.addr.ipv6, addr_str)) {
+        puts("Error parsing destination address");
+        return -1;
+    }
+
+    if (sock_udp_create(&udp_sock, &local, NULL, 0) < 0) {
+        puts("Error creating UDP sock");
+        return -1;
+    }
+
+    if (sock_dtls_create(&dtls_sock, &udp_sock,
+                        SOCK_DTLS_CLIENT_TAG,
+                        SOCK_DTLS_1_2, SOCK_DTLS_CLIENT) < 0) {
+        puts("Error creating DTLS sock");
+        sock_udp_close(&udp_sock);
+        return -1;
+    }
+
+#ifdef DTLS_PSK
+    credman_credential_t credential = {
```

Comment:
How about making `credential` a static const and moving to the global variables. This way, you can leave the `#ifdef` out of this function.

Reply:
I'll change that.